downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md

```diff
@@ -54,10 +54,67 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 10 (2201.10.0)
 
 ## Runtime updates
 
-### New features
-
 ### Improvements
 
+#### Added support for the `shift`, `unshift`, `pop`, and `remove` functions with tuples
+
+The runtime now supports the `shift`, `unshift`, `pop`, and `remove` operations on tuples similar to arrays, as long as they do not violate inherent type or mutability constraints.
+
+The following are now allowed.
+
+```ballerina
+[int, int...] tuple1 = [1, 2];
+int val1 = tuple1.shift();          // 1
+
+[string, string...] tuple2 = ["hello"];
+tuple2.unshift("world", "cat");     // ["hello", "world", "cat"]
+
+[int, string, int...] tuple3 = [1, "hello", 4];
+var val3 = tuple3.pop();            // 4
```

Comment:
Thought of keeping the examples simple instead of directing attention to the type of `val3`. The type of `val3` will be inferred anyways when `var` is used right? Can change the type to `int|string` if that is better but for some examples it can be quite long such as `int|string|float`. Can go with `anydata` also.  What do you think?

Reply:
I would use the proper union type, even though it may be long, since it gives a better idea of what exactly happens and how type takes into account the different member types of a tuple.